receiver/awscontainerinsightreceiver/internal/k8sapiserver/leaderelection.go

```diff
@@ -65,8 +65,10 @@ func WithLeaderLockUsingConfigMapOnly(leaderLockUsingConfigMapOnly bool) LeaderE
 
 func NewLeaderElection(logger *zap.Logger, options ...LeaderElectionOption) (*LeaderElection, error) {
 	le := &LeaderElection{
-		logger:      logger,
-		k8sClient:   k8sclient.Get(logger),
+		logger: logger,
+		k8sClient: k8sclient.Get(logger,
+			k8sclient.CaptureOnlyNodeLabelsInfo(true),
```

Comment:
This should be empty if the node is not hyperpod (instance type doesn't start with ml)

Reply:
I don't have direct access to captureOnlyNodeLabelInfo from leaderElection. also i don't see the CaptureNodeLevelInfo getting tested. any suggestions?

Reply:
>so this will be always true and not contorlled by a flag or something?

Yes this will run only on the leader agent to collect the labels for all nodes.